### PR TITLE
Remove direct hashlib.sha256() calls

### PR DESCRIPTION
**Pull request by jnewbery**
Makes notebooks easier by using the sha256() convenience function.

Simplification from workshop run 2.

### CODE REVIEW

File: 1.1-schnorr-signatures.ipynb [hunk 1]
```diff
@@ -6,10 +6,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import hashlib\n",
     "import random\n",
     "import util\n",
-    "from test_framework.key import generate_key_pair, ECKey, ECPubKey, jacobi_symbol, SECP256K1_FIELD_SIZE, SECP256K1_ORDER"
+    "from test_framework.key import generate_key_pair, ECKey, ECPubKey, jacobi_symbol, SECP256K1_FIELD_SIZE, SECP256K1_ORDER\n"
```
Comment (bitschmidty) [inline 1.1-schnorr-signatures.ipynb]: Missing comma at the end of line causing error loading notebook

File: 2.2-taptweak.ipynb [hunk 2]
```diff
@@ -6,7 +6,6 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import hashlib\n",
     "from io import BytesIO\n",
     "import random\n",
     "\n",
```
Comment (bitschmidty) [inline 2.2-taptweak.ipynb]: Missed import of sha256 on `from test_framework.messages` line

File: 2.3-tapscript.ipynb [hunk 3]
```diff
@@ -432,7 +431,7 @@
     "print(\"pubkey0: {}\".format(pubkey0.get_bytes().hex()))\n",
     "print(\"pubkey1: {}\\n\".format(pubkey1.get_bytes().hex()))\n",
     "\n",
-    "# Method: 32B preimage - hashlib.new('sha256', bytes).digest()\n",
+    "# Method: 32B preimage - sha256('sha256', bytes)\n",
     "# Method: 20B digest - hashlib.new('ripemd160', bytes).digest()\n",
```
Comment (bitschmidty) [inline 2.3-tapscript.ipynb]: Still need hashlib for this ripemd160 call

File: 1.1-schnorr-signatures.ipynb (hunk 1 above)
Comment (jnewbery) [reply 1.1-schnorr-signatures.ipynb]: fixed

File: 2.2-taptweak.ipynb (hunk 2 above)
Comment (jnewbery) [reply 2.2-taptweak.ipynb]: fixed

File: 2.3-tapscript.ipynb (hunk 3 above)
Comment (jnewbery) [reply 2.3-tapscript.ipynb]: fixed

**Comment by jnewbery** [comment]
All comments addressed. Thanks for the review @bitschmidty

**Pull request merged**